components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/ApplicationAuthenticationService.java

```diff
@@ -43,7 +45,8 @@ public ApplicationAuthenticator getAuthenticator(String name) throws Application
 
         ApplicationAuthenticator appAuthenticator = null;
 
-        for (ApplicationAuthenticator authenticator : FrameworkServiceComponent.getAuthenticators()) {
+        for (ApplicationAuthenticator authenticator :
+                ApplicationAuthenticatorManager.getInstance().getSystemDefinedAuthenticators()) {
```

Comment:
IMO for this service we should be able to reply on the property that's in the authenticator which denotes whether it's api based supported or not

Reply:
This is service is only using for API based authentication ATM. As we are not supporting that for user defined authenticator, have not improves this method.

Reply:
but is checks if api auth is enabled. Then at the point we support it's just a matter of enabling that support from adapter